utils/symbolicate-linux-fatal

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+# symbolicate-linux-fatal - Symbolicate Linux stack traces -*- python -*-
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+#
+# ----------------------------------------------------------------------------
+#
+# Symbolicates fatalError stack traces on Linux. Takes the main binary
+# and a log file containing a stack trace. Non-stacktrace lines are output
+# unmodified. Stack trace elements are analysed using reconstructed debug
+# target matching the original process in where shared libs where mapped.
+#
+# TODOs:
+# * verbose output
+# * search symbols by name for the not <unavailable> ones
+#
+# ----------------------------------------------------------------------------
+
+from __future__ import print_function
+
+import argparse
+import sys
+import lldb
+import subprocess
+
+DYN_LIBS = {}
+BINARY = None
+
+
+def process_ldd(lddoutput):
+    for line in lddoutput.splitlines():
+        ldd_tokens = line.split()
+        if len(ldd_tokens) >= 3:
+            DYN_LIBS[ldd_tokens[0]] = ldd_tokens[2]
+
+
+def create_lldb_target(memmap):
+    lldb_debugger = lldb.SBDebugger.Create()
+    lldb_target = lldb_debugger.CreateTargetWithFileAndArch(
+        BINARY, lldb.LLDB_ARCH_DEFAULT)
+    module = lldb_target.GetModuleAtIndex(0)
+    # lldb seems to treat main binary differently, slide offset must be zero
+    lldb_target.SetModuleLoadAddress(module, 0)
+    for dynlib_path in memmap:
+        if BINARY not in dynlib_path:
+            module = lldb_target.AddModule(
+                dynlib_path, lldb.LLDB_ARCH_DEFAULT, None, None)
+            lldb_target.SetModuleLoadAddress(module, memmap[dynlib_path])
+    return lldb_target
+
+
+def process_stack(stack):
+    if len(stack) == 0:
+        return
+    memmap = {}
+    full_stack = []
+    for line in stack:
+        stack_tokens = line.split()
+        dynlib_fname = stack_tokens[1]
+        if dynlib_fname in DYN_LIBS:
+            dynlib_path = DYN_LIBS[dynlib_fname]
+        else:
+            if dynlib_fname in BINARY:
+                dynlib_path = BINARY
+            else:
+                dynlib_path = None
+
+        if "<unavailable>" in stack_tokens[3]:
+            framePC = int(stack_tokens[2], 16)
+            symbol_offset = int(stack_tokens[-1], 10)
+            dynlib_baseaddr = framePC - symbol_offset
+            if dynlib_path in memmap:
+                if memmap[dynlib_path] != dynlib_baseaddr:
+                    print("Mismatched base address for: {0:s}, "
+                          "had: {1:x}, now got {2:x}".format(
+                            dynlib_path, memmap[dynlib_path], dynlib_baseaddr))
+                    sys.exit(1)
```

Comment:
This isn't necessarily better than your approach, but an alternative to printing and calling `sys.exit()` here would be to throw an exception, such as `throw Error("Mismatched ...")`. It's fewer lines of code, and emits a backtrace, but maybe you don't want users to see that. Up to you!